### PR TITLE
storing flow-config in atom and files in subdir

### LINKED ISSUE #123: Changes in flow-config require manual intervention
flow-api should require no manual intervention when the flow configuration changes, as other services.

### PR DESCRIPTION
**Pull request by dlebrero**
We store the p12 and appengine.xml file in disk, so when there is a
 refresh, they need to be refetched.

We opted to store the files in a subdirectory with the schema
$tmpdir/$refresh-timestamp, so that if there is another thread using the
old ones, it cannot be impacted by the refresh.

A background thread cleans up old subdirectories after 60 secs.

Added also a docker-compose environment

Fixes #123

### CODE REVIEW

File: api/Dockerfile-dev [hunk 1]
```diff
@@ -0,0 +1,6 @@
+FROM clojure:lein-2.7.1
```
Comment (iperdomo) [inline api/Dockerfile-dev]: `clojure:lein-2.7.1` -> `clojure:lein-2.8.1`

File: api/dev/setup-dev-user-in-container.sh [hunk 2]
```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env sh
```
Comment (iperdomo) [inline api/dev/setup-dev-user-in-container.sh]: We don't need this approach anymore, check the other projects... `run-as-user.sh`

**Comment by iperdomo** [comment]
> We opted to store the files in a subdirectory with the schema $tmpdir/$refresh-timestamp, so that if there is another thread using the old ones, it cannot be impacted by the refresh.

In the `POST` request we get, we have sha1 of the commit so we could use that instead?

https://developer.github.com/v3/activity/events/types/#pushevent

What do you think?

File: api/dev/setup-dev-user-in-container.sh (hunk 2 above)
Comment (dlebrero) [reply api/dev/setup-dev-user-in-container.sh]: The "run-as-user" approach doesnt work. For some reason groupmod returns an error, which means that we need to use groupadd, which means that we need to check if the group exists before calling groupadd, which means that we will end with something similar to the "setup-dev-user"

File: api/Dockerfile-dev (hunk 1 above)
Comment (dlebrero) [reply api/Dockerfile-dev]: fixed

**Comment by dlebrero** [comment]
I think using the sha is technically more correct, but I cannot see it fixing any issue with the timestamp approach.

Using the sha would mean to also find the sha on startup, which will add some additional complexity on top of having to parse the github request.

IMO, not worth the effort.

**Comment by iperdomo** [comment]
By changing to lein-2.8.1 and clearing the maven cache, i'm getting this error:

    Tried to use insecure HTTP repository without TLS
    This is almost certainly a mistake; however in rare cases where it's intentional please see `lein help faq` for details.

**Pull request merged**